Update Kubeadm with off line configurations (#4073)
cri-o/cri-o: tutorials/kubeadm.md

Given you've set CIDR, and you've properly set the kubelet file, all you need to do is start crio (as defined [here](setup.md)), and run:
`kubeadm init --pod-network-cidr=$CIDR`

> # Running kubeadm in a off line network

Reviewer: ```suggestion
# Running kubeadm in an off line network
```

> We will assume that the user has installed CRI-O and alls necessary packages. We will also assume that all necessary components are configured and everything is working as expected. The user should have a private repo where the docker images are pushed. Sample of images fot version 1.18.2:

Reviewer 2: s/fot/for

Reviewer: Nice catch, missed that.

Author: Done

```bash
$ kubeadm config images list --image-repository user.private.repo --kubernetes-version=v1.18.2
user.private.repo/kube-apiserver:v1.18.2
user.private.repo/kube-controller-manager:v1.18.2
user.private.repo/kube-scheduler:v1.18.2
user.private.repo/kube-proxy:v1.18.2
user.private.repo/pause:3.2
user.private.repo/etcd:3.4.3-0
user.private.repo/coredns:1.6.7
```

The user needs to configure the [registries.conf](https://www.mankier.com/5/containers-registries.conf) file.

Sample of configurations:

```bash
$ cat /etc/containers/registries.conf
[[registry]]

> prefix = "k8s.gcr.io/pause:3.2"

Reviewer 3: We regex here so instead of one image you could cover all them in the example.

Author: The images differ from each other. I am not sure if one regex can match all in once. On the other hand I am not that good with regex. Can you propose a sample and we can take a look?

insecure = false
blocked = false
location = "user.private.repo/pause:3.2"
```

> Next the user should reload and restart CRI-O service to load the configurations.

Reviewer: ```suggestion
Next the user should reload and restart the CRI-O service to load the configurations.
```

> Last step the user also should add the cgroup driver on the configuration file or pass it as a parameter when starting kubeadm for the master node sample of parameters for configuration file:

Reviewer: ```suggestion
Last step. The user also should add the cgroup driver to the configuration file or pass it as a parameter when starting kubeadm for the master node.  The following are samples of parameters for the configuration file:
```

> Last step. The user also should add the cgroup driver to the configuration file or pass it as a parameter when starting kubeadm for the master node.  The following are samples of parameters for the configuration file:

Reviewer 2: I would drop the `Last step`

Author: I can remove it but this is an important part to make it work. I had difficulties starting with the cri-o socket without this step but it up to you guys to decide. It is also documented in the official documentation [Configure cgroup driver used by kubelet on control-plane node](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node). If it more preferred I can make a reference?

Reviewer 2: ah no I meant remove the string `Last Step.` and keep `The user also should add the cgroup driver to the configuration file or pass it as a parameter when starting kubeadm...`

Author: To be updated in a few hours.

> Last step. The user also should add the cgroup driver to the configuration file or pass it as a parameter when starting kubeadm for the master node.  The following are samples of parameters for the configuration file:

Reviewer 2: between the second sentence (`...master node`) and the last, there are two spaces. can we drop to one please

Author: Done
